model-engine/model_engine_server/core/celery/celery_autoscaler.py

```diff
@@ -0,0 +1,633 @@
+from kubernetes_asyncio.client.rest import ApiException
+from kubernetes_asyncio.config.config_exception import ConfigException
+from model_engine_server.core.aws.roles import session
+from model_engine_server.core.celery import (
+    TaskVisibility,
+    celery_app,
+    get_all_db_indexes,
+    get_redis_host_port,
+    inspect_app,
+)
+from model_engine_server.core.loggers import logger_name, make_logger
+
+
+def excluded_namespaces():
+    try:
+        from plugins.celery_autoscaler_dependencies import CELERY_AUTOSCALER_EXCLUDED_NAMESPACES
+
+        return CELERY_AUTOSCALER_EXCLUDED_NAMESPACES
+    except ModuleNotFoundError:
+        return []
+
+
+ELASTICACHE_REDIS_BROKER = "redis-elasticache-message-broker-master"
+SQS_BROKER = "sqs-message-broker-master"
+
+UPDATE_DEPLOYMENT_MAX_RETRIES = 10
+
+SQS_SAMPLE_COUNT = 10
+
+logger = make_logger(logger_name())
+
+autoscaler_broker = os.environ.get("BROKER_NAME", SQS_BROKER)
+aws_profile = os.environ.get("AWS_PROFILE")
+
+
+@dataclasses.dataclass
+class CeleryAutoscalerParams:
+    queue: str
+    broker: str = SQS_BROKER
+    task_visibility: TaskVisibility = TaskVisibility.VISIBILITY_1H
+    per_worker: int = 1
+    min_workers: int = 0
+    max_workers: int = 1
+
+
+def _hash_any_to_int(data: Any):
+    return int(hashlib.md5(str(data).encode()).hexdigest(), 16)
+
+
+async def list_deployments(core_api, apps_api) -> Dict[Tuple[str, str], CeleryAutoscalerParams]:
+    namespaces = await core_api.list_namespace()
+    celery_deployments_params = {}
+    for namespace in namespaces.items:
+        namespace_name = namespace.metadata.name
+        if namespace_name in excluded_namespaces():
+            continue
+        namespace_start_time = time.time()
+        deployments = await apps_api.list_namespaced_deployment(namespace=namespace_name)
+        logger.info(
+            f"list_namespaced_deployment with {namespace_name} took {time.time() - namespace_start_time} seconds"
+        )
+        for deployment in deployments.items:
+            deployment_name = deployment.metadata.name
+            annotations = deployment.metadata.annotations
+
+            if not annotations:
+                continue
+
+            # Parse parameters
+            params = {}
+
+            if "celery.scaleml.autoscaler/broker" in annotations:
+                deployment_broker = annotations["celery.scaleml.autoscaler/broker"]
+            else:
+                deployment_broker = ELASTICACHE_REDIS_BROKER
+
+            if deployment_broker != autoscaler_broker:
+                logger.debug(
+                    f"Skipping deployment {deployment_name}; deployment's broker {deployment_broker} is not {autoscaler_broker}"
+                )
+                continue
+
+            for f in dataclasses.fields(CeleryAutoscalerParams):
+                k = f.name
+                # TODO: this technically can remain the same, since Celery deployments won't have the `channel` field and
```

Comment:
I guess we can remove this comment probably? there is no NATS autoscaler anymore